### PR TITLE
Refactor indentation

### PR DESCRIPTION
**Pull request by stojanovic**
Refactor mentioned in #552:

- Adds indentation for props in new lines
- Preserves 80 chars limit

Only exception is [this line](https://github.com/koajs/koa/blob/master/test/application/respond.js#L597) because it'll be exactly 80 chars after #553 is merged to master.

### CODE REVIEW

**Comment by broucz** [comment]
Not successful coverage will be fix with #550 merge.

**Comment by stojanovic** [comment]
Perfect, I'll wait for that then :)

**Comment by jonathanong** [comment]
similarly to #550, are there relevant eslint rules?

**Comment by stojanovic** [comment]
@jonathanong rules added. I just don't know if the rule for indented properties exist at all.
But we now have rules for dot location and 80 chars line length.

File: .eslintrc [hunk 1]
```diff
@@ -11,3 +11,6 @@ rules:
   yoda: 0
   arrow-parens: [2, "as-needed"]
   arrow-spacing: 2
+  dot-location: [2, "property"]
+  "max-len": [2, 80, 4, {"ignoreComments": true, "ignoreUrls": true, "ignorePattern": "^\\s*var\\s.+=\\s*require\\s*\\("}]
```
Comment (broucz) [inline .eslintrc]: May you forget to remove parentheses around `max-len` no ?

Comment (stojanovic) [inline .eslintrc]: @broucz ha, you are right :)
Fixed, I also simplified that rule a bit, so it has less than 80 chars long :)

Comment (broucz) [inline .eslintrc]: if even the rule is < 80: nice ;)

**Comment by jonathanong** [comment]
needs a rebase =/

do you also mind squashing your commits so you don't have any like "fix merge conflicts"? otherwise, i'll just squash them all into one

**Comment by stojanovic** [comment]
Fixed and I squashed all into one since it makes sense to have just one commit here.
Next time I'll do this immediately.

File: test/request/accepts.js [hunk 2]
```diff
@@ -8,8 +8,14 @@ describe('ctx.accepts(types)', function(){
     describe('when Accept is populated', function(){
       it('should return all accepted types', function(){
         const ctx = context();
-        ctx.req.headers.accept = 'application/*;q=0.2, image/jpeg;q=0.8, text/html, text/plain';
-        ctx.accepts().should.eql(['text/html', 'text/plain', 'image/jpeg', 'application/*']);
+        ctx.req.headers.accept =
```
Comment (jonathanong) [inline test/request/accepts.js]: not sure i like this... weird IMO lol

Comment (stojanovic) [inline test/request/accepts.js]: It is indeed weird, but I don't know hot to compress it in 80 chars better :D

Comment (tejasmanohar) [inline test/request/accepts.js]: @stojanovic you may have misunderstood my point earlier

`ctx.accepts('text/html', 'text/plain', 'image/jpeg', 'application/*').should.equal('text/html');` is fine all in 1 line. it's just that if it's getting _too long_... then you can do something like

```javascript
ctx
  .accepts('text/html', 'text/plain', 'image/jpeg', 'application/*')
  .should.equal('text/html')
```

indenting -> chaining one function but not the others is odd.. so yea, what it currently is seems weird to me too lol @jonathanong

Comment (jonathanong) [inline test/request/accepts.js]: anyone actually want to limit to 80 characters? i personally don't mind long lines as long as it's just a really long constant/literal

Comment (stojanovic) [inline test/request/accepts.js]: I can remove that limit and rebase if that works better, it'll definitely be a bit less weird.
@tejasmanohar ok, that's easy fix too.

**Comment by felixfbecker** [comment]
The refactor on indentation is great, but the forced 80 column limit actually makes stuff _harder_ to read. We all use 16:9 monitors and our editors support soft wrap. 80 columns was a convention for 4:3 cathode ray tubes running `vi`.

**Comment by tj** [comment]
haha agreed, same argument as tabs being better, your editor can display whatever it likes

**Comment by tejasmanohar** [comment]
if it gets way way too long or a big chain of functions, then i'd prefer the
```javascript
obj
  .x(...)
  .y(...)
  .z(...)
  .a(...)
```
format just for readability, but i don't care about arbitrary hard limits like 80 chars. just word wrap in your editor if you'd like :)

File: test/request/accepts.js (hunk 2 above)
Comment (tejasmanohar) [inline test/request/accepts.js]: yep, i don't care about hard limits on chars / line

**Comment by felixfbecker** [comment]
Agree with @tejasmanohar!

**Comment by stojanovic** [comment]
It should be a bit better now, I removed 80 chars limit and weird line breaks

**Comment by jonathanong** [comment]
@stojanovic you gotta rebase master, not merge master!

**Pull request closed without merging**